src/docs/guides/astro.md

```diff
@@ -0,0 +1,231 @@
+
+A new Astro app will be provisioned for you in the `blog` directory.
+
+### Run the Astro App locally
+
+Enter your project directory using `cd blog`.
+
+Start the local dev server by running the following command:
+
+```bash
+npm run dev
+```
+
+Open your browser and go to `http://localhost:4321` to see your app.
+
+### Enable Server side rendering (SSR)
+
+Astro has several [SSR adapters](https://docs.astro.build/en/guides/server-side-rendering/). These adapters are used to run your project on the server and handle SSR requests.
+
+Let's add the Node adapter to enable SSR in our blog project.
+
+Run the command below in your terminal:
+
+```bash
+npx astro add node
+```
+
+Select **Yes** at the prompt to proceed. The Node adapter will be installed, and our Astro config file will be updated accordingly.
+
+Open up the `astro.config.mjs` file:
+
+```js
+// @ts-check
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+
+import sitemap from '@astrojs/sitemap';
+
+import node from '@astrojs/node';
+
+// https://astro.build/config
+export default defineConfig({
+  site: 'https://example.com',
+  integrations: [mdx(), sitemap()],
+  output: 'server',
+
+  adapter: node({
+    mode: 'standalone',
+  }),
+});
+```
+
+In the config file, `output` is set to `server`, meaning every page in the app is server-rendered by default.
+
+For mostly static sites, set `output` to `hybrid`. This allows you to add `export const prerender = false` to any file that needs to be server-rendered on demand.
+
+## Deploy the Astro App to Railway
+
+Railway offers multiple ways to deploy your Astro app, depending on your setup and preference. 
+
+### One-Click Deploy from a Template
+
+If you’re looking for the fastest way to get started, the one-click deploy option is ideal.
+
+Click the button below to begin:
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/Ic0JBh)
+
+We highly recommend that [you eject from the template after deployment](/guides/deploy#eject-from-template-repository) to create a copy of the repo on your GitHub account.
+
+**Note:** You can also choose from a <a href="https://railway.app/templates?q=astro" target="_blank">variety of Astro app templates</a> created by the community.
+
+### Deploy from the CLI
+
+1. **Install the Railway CLI**:
+    - <a href="/guides/cli#installing-the-cli" target="_blank">Install the CLI</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate it</a> using your Railway account.
+2. **Initialize a Railway Project**:
+    - Run the command below in your Vue app directory. 
+        ```bash
+        railway init
+        ```
+    - Follow the prompts to name your project.
+    - After the project is created, click the provided link to view it in your browser.
+3. **Modify your app's start script and Astro config**:
```

Comment:
Interestingly, I did that before. But I moved into later into the deploy section because I believed you start to worry about that when you are about to deploy 🤔

Reply:
Maybe some kind of prerequisites section?

Reply:
I will move it back to before the deploy section. I see the benefit of placing it before both sections.